Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalMemoryReaderTests.swift

```diff
@@ -7,58 +7,90 @@
 import XCTest
 @testable import Datadog
 
+private class Allocation {
+    fileprivate let numberOfBytes: Int
+    private var pointer: UnsafeMutablePointer<UInt8>! // swiftlint:disable:this implicitly_unwrapped_optional
+
+    init(numberOfBytes: Int) {
+        self.numberOfBytes = numberOfBytes
+    }
+
+    func allocate() {
+        pointer = UnsafeMutablePointer<UInt8>.allocate(capacity: numberOfBytes)
+        pointer.initialize(repeating: 0, count: numberOfBytes)
+    }
+
+    func deallocate() {
+        pointer.deallocate()
+    }
+}
+
 internal class VitalMemoryReaderTest: XCTestCase {
+    private let allocation = Allocation(
+        numberOfBytes: 8 * 1_024 * 1_024 // 8MB is significantly more than any other allocation in tests process
```

Comment:
The biggest known allocation in tests is `[NSThread start]`: `512KiB`. Previously, we were using `~1MB` which was causing flakiness:

<img width="874" alt="Screenshot 2021-09-10 at 14 12 49" src="https://user-images.githubusercontent.com/2358722/133263580-ae54d549-093a-4be0-b144-d142191409fb.png">

Now with `8MB` it should be more stable.